test/acceptance/features/steps/steps.py

```diff
@@ -303,24 +307,34 @@ def quarkus_app_is_imported_as_knative_service(context, application_name):
 
 
 # STEP
-
-
 @then(u'"{app_name}" deployment must contain SBR name "{sbr_name}"')
 def then_envFrom_contains(context, app_name, sbr_name):
     time.sleep(60)
     openshift = Openshift()
+    secret = ""
+    sbr = ServiceBinding()
+    if "sbr_name" in context:
+        json_path = "{.status.secret}"
+        secret = sbr.get_servicebinding_info_by_jsonpath(context.sbr_name, context.namespace.name, json_path)
+
     result = openshift.get_deployment_envFrom_info(app_name, context.namespace.name)
     # Expected result from 'oc' (openshift client) v4.5
-    expected_result_oc_45 = f'secretRef:map[name:{sbr_name}]'
+    expected_result_oc_45 = f'secretRef:map[name:{secret}]'
     # Expected result from 'oc' (openshift client) v4.6+
-    expected_result_oc_46 = f'{{"secretRef":{{"name":"{sbr_name}"}}}}'
+    expected_result_oc_46 = f'{{"secretRef":{{"name":"{secret}"}}}}'
     assert re.search(re.escape(expected_result_oc_45), result) is not None or re.search(re.escape(expected_result_oc_46), result) is not None, \
-        f'\n{app_name} deployment should contain secretRef: {sbr_name} \nActual secretRef: {result}'
+        f'\n{app_name} deployment should contain secretRef: {secret} \nActual secretRef: {result}'
 
 
 # STEP
-@then(u'deployment must contain intermediate secret "{intermediate_secret_name}"')
-def envFrom_contains_intermediate_secret_name(context, intermediate_secret_name):
+@then(u'deployment must contain intermediate secret')
+def envFrom_contains_intermediate_secret_name(context):
+    time.sleep(10)
```

Comment:
remove absolute waiting, use polling instead.

Reply:
Done here b7139c7

Reply:
@pedjak I have added `time.sleep` here -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-28308d773b6915650013e5b2b3dba17f37a5dda1b20bd150ee56401e72b470b2R321

because polling retrieved the old SBR.Status.Secret value